Fix: New Goo validators

syphax-bouazzouni opened the pull request:
Fix https://github.com/ncbo/goo/pull/139#issuecomment-1699645273

mdorf commented:
I already implemented similar fixes, so probably no need for the PR. For the `user.rb` `to_s` method, my solution was almost the same as yours:
```
      def to_s
        if self.bring?(:username)
          LinkedData::Utils::Triples.last_iri_fragment self.id.to_s
        else
          self.username.to_s
        end
      end
```
For the rest of the failing tests, can you elaborate why (reload=true) is no longer needed and no longer works?

syphax-bouazzouni commented:
Yes, your version using `LinkedData::Utils::Triples.last_iri_fragment ` is better. 

As the why I removed `(reload=true)`, frankly, I don't really remember. At that time having the `reload` enabled broke our tests, but retesting today seems to be no longer the case.  

So I updated this to use your version of the fix and removed the change related to `(reload=true)`.

But to go more in deep about this story of `(reload=true)`, do you know in what is it used? I did some research in the code but it is used only in the tests (Agroportal version at least). And removing did not affect the tests
